### PR TITLE
EDX-4464 Add DiscoverComponents method for xrtClient

### PR DESCRIPTION
**Pull request by lindseysimple**
If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-messaging/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,11 +6,11 @@ require (
 	github.com/eclipse/paho.mqtt.golang v1.4.2
 	github.com/edgexfoundry/go-mod-core-contracts/v2 v2.3.0
 	github.com/go-redis/redis/v7 v7.3.0
-	github.com/google/uuid v1.3.0
+	github.com/google/uuid v1.4.0
```
Comment (cloudxxx8) [inline go.mod]:
```suggestion
	github.com/google/uuid v1.5.0
```

File: clients/xpert/xrt.go [hunk 2]
```diff
@@ -177,6 +177,30 @@ func (c *xrtClient) sendXrtRequestWithTimeout(ctx context.Context, requestTopic
 	return nil
 }
 
+// sendXrtRequestWithSubTimeout publish the xrt request and wait for responses from multiple xrt nodes for the specific subscribe timeout
+func (c *xrtClient) sendXrtRequestWithSubTimeout(ctx context.Context, requestTopic string, requestId string, request interface{},
+	response interface{}, subscribeTimeout time.Duration) errors.EdgeX {
```
Comment (cloudxxx8) [inline clients/xpert/xrt.go]:
```suggestion
	response any, subscribeTimeout time.Duration) errors.EdgeX {
```

File: clients/xpert/xrt.go [hunk 3]
```diff
@@ -177,6 +177,30 @@ func (c *xrtClient) sendXrtRequestWithTimeout(ctx context.Context, requestTopic
 	return nil
 }
 
+// sendXrtRequestWithSubTimeout publish the xrt request and wait for responses from multiple xrt nodes for the specific subscribe timeout
+func (c *xrtClient) sendXrtRequestWithSubTimeout(ctx context.Context, requestTopic string, requestId string, request interface{},
```
Comment (cloudxxx8) [inline clients/xpert/xrt.go]:
```suggestion
func (c *xrtClient) sendXrtRequestWithSubTimeout(ctx context.Context, requestTopic string, requestId string, request any,
```

File: clients/xpert/xrtcomponent.go [hunk 4]
```diff
@@ -27,3 +28,20 @@ func (c *xrtClient) UpdateLuaScript(ctx context.Context, luaScript string) error
...
+	var response []xrtmodels.MultiComponentsResponse
+
+	err := c.sendXrtRequestWithSubTimeout(ctx, c.requestTopic, request.RequestId, request, &response, subscribeTimeout)
+	if err != nil {
+		return nil, errors.NewCommonEdgeX(errors.Kind(err), "failed to discover the xrt components", err)
+	}
+
+	for _, compResp := range response {
+		if compResp.Result.Error() != nil {
+			return nil, compResp.Result.Error()
```
Comment (cloudxxx8) [inline clients/xpert/xrtcomponent.go]: ignore the individual component result error, so the for loop should be removed
the caller should handle those errors

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (lindseysimple) [reply go.mod]: Done

File: clients/xpert/xrt.go (hunk 2 above)
Comment (lindseysimple) [reply clients/xpert/xrt.go]: Done

File: clients/xpert/xrtcomponent.go (hunk 4 above)
Comment (lindseysimple) [reply clients/xpert/xrtcomponent.go]: Done

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**